### PR TITLE
Improving robustness[WIP]

### PR DESCRIPTION
**Pull request by JeanMarcR**
- adding mutex lock/unlock
- removing padding characters

### CODE REVIEW

File: ISM43362/ISM43362.cpp [hunk 1]
```diff
@@ -175,18 +175,6 @@ bool ISM43362::check_response(void)
         return false;
     }
 
-    /*  Inventek module do stuffing / padding of data with 0x15,
```
Comment (LMESTM) [inline ISM43362/ISM43362.cpp:178]: Does it mean that 0x15 padding has been replaced by 0x3E padding in more recent Inventek firmware version ? 
In the specification I found, we still see references to 0x15 padding ?

**Review by LMESTM — COMMENTED** [review]
Just one question: the Inventek firmware spec that I found still refers to 0x15 padding, so how do we come the 0x3E padding ? 
0x3E is '>' and part of the AT protocol. Shouldn't 0x15 padding management maintained as well or do you consider this will be handled somewhere else ?

**Comment by jeromecoutant** [comment]
Let's update #90?

Or you would like to use #90 for mutex lock/unlock
and #91 for padding issues ?

**Comment by JeanMarcR** [comment]
@jeromecoutant The PR(#90) is updated.

I would like to close the current PR(#91).
Do you agree ?

File: ISM43362/ISM43362.cpp (hunk 1 above)
Comment (JeanMarcR) [reply ISM43362/ISM43362.cpp:178]: @LMESTM
The last useful character is ">" -> ASCII (0x3E).
So, I check the presence of ">" and I flush.
But after the ">" there are useless characters (0x15) which were previously removed
Please, is it possible to continue this conversation in PR (#90) which is up to date now.

**Pull request closed without merging**